File: 0735-asteroid-collision/0735-asteroid-collision.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        stack = [-1]
        for a in asteroids:
            while stack[-1] > 0 and a < 0:
                diff = stack[-1] + a
                if diff == 0:
                    a = 0
                    stack.pop()
                elif diff > 0:
                    a = 0
                elif diff < 0:
                    stack.pop()
            if a != 0:
                stack.append(a)
        return stack[1::]

File: 0735-asteroid-collision/0735-asteroid-collision.js
/**
 * @param {number[]} asteroids
 * @return {number[]}
 */
var asteroidCollision = function(asteroids) {
    const stack = [-1]
    for (let a of asteroids) {
        while (a < 0 && stack.at(-1) > 0) {
            const diff = stack.at(-1) + a
            if (diff > 0) {
                a = 0
            } else if (diff < 0) {
                stack.pop()
            } else {
                a = 0
                stack.pop()
            }
        }
        if (a !== 0) {
            stack.push(a)
        }
    }
    return stack.slice(1)
};